tests/syscall/mount-overlayfs.bats

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env bats
+
+#
+# Verify trapping & emulation on "mount" and "unmount2" syscalls for overlayfs mounts
+#
+
+
+load ../helpers/run
+load ../helpers/syscall
+load ../helpers/docker
+load ../helpers/sysbox-health
+
+ovs_path="/var/lib/docker/overlay2"
+
+function teardown() {
+  sysbox_log_check
+}
+
+# Verify basic overlayfs mount instruction.
+@test "mount overlayfs basic" {
+
+  local lower_path="$ovs_path"/root/lower
+  local upper_path="$ovs_path"/root/upper
+  local work_path="$ovs_path"/root/work
+  local merge_path="$ovs_path"/root/merge
+
+  local syscont=$(docker_run --rm nestybox/alpine-docker-dbg:latest tail -f /dev/null)
+
+  # Prepare mount setup.
+  docker exec "$syscont" bash -c \
+    "mkdir -p $lower_path $upper_path $work_path $merge_path"
+  [ "$status" -eq 0 ]
+
+  # Mount overlayfs at $merge_path.
+  docker exec "$syscont" bash -c \
+    "mount -t overlay overlay -olowerdir=$lower_path,upperdir=$upper_path,workdir=$work_path $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_mnt $syscont $merge_path
+
+  # Umount overlayfs mountpoint.
+  docker exec "$syscont" bash -c "umount $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_umnt $syscont $merge_path
+
+  docker_stop "$syscont"
+}
+
+# Verify fs addition operations.
+@test "mount overlayfs addition ops" {
+
+  local lower_path="$ovs_path"/root/lower
+  local upper_path="$ovs_path"/root/upper
+  local work_path="$ovs_path"/root/work
+  local merge_path="$ovs_path"/root/merge
+
+  local lower_dir="$ovs_path"/root/lower/lower_dir
+  local lower_file="$ovs_path"/root/lower/lower_file
+  local upper_dir="$ovs_path"/root/upper/upper_dir
+  local upper_file="$ovs_path"/root/upper/upper_file
+
+  local syscont=$(docker_run --rm nestybox/alpine-docker-dbg:latest tail -f /dev/null)
+
+  docker exec "$syscont" bash -c \
+    "mkdir -p $lower_path $upper_path $work_path $merge_path && \
+     mkdir -p $lower_dir $upper_dir && \
+     touch $lower_file $upper_file"
+  [ "$status" -eq 0 ]
+
+  # Mount overlayfs at $merge_path.
+  docker exec "$syscont" bash -c \
+    "mount -t overlay overlay -olowerdir=$lower_path,upperdir=$upper_path,workdir=$work_path $merge_path"
+  [ "$status" -eq 0 ]
+
+  verify_syscont_overlay_mnt $syscont $merge_path
+
+  # Verify dir/file content in merge folder is the expected one.
+  docker exec "$syscont" stat $lower_dir $lower_file $upper_dir $upper_file
+  [ "$status" -eq 0 ]
+
+  # Add new dir.
+  docker exec "$syscont" bash -c "mkdir $merge_path/new_dir"
+  [ "$status" -eq 0 ]
+
+  # Add new file.
+  docker exec "$syscont" bash -c "touch $merge_path/new_file"
+  [ "$status" -eq 0 ]
+
+  # Confirm new dir/file is present.
+  docker exec "$syscont" bash -c "stat $merge_path/new_dir $merge_path/new_file"
```

Comment:
Let's also verify that the new file is present in the upper dir.

Reply:
Wondering why are you suggesting this. If new file is not present in the upper dir, then we would be dealing with a kernel bug, not a sysbox one.

Reply:
yes you are right, but it's a sanity check to make sure we are not messing up anything in the way sysbox processes the ovfs mount.

Reply:
Oh i see what you mean: add files, folders, and files-within-folders. Sure will add this to the next testcase.

Reply:
i did it anyways.